internal/archive/archive.go

```diff
@@ -116,6 +147,57 @@ func (a *ubuntuArchive) Fetch(pkg string) (io.ReadCloser, error) {
 
 const ubuntuURL = "http://archive.ubuntu.com/ubuntu/"
 const ubuntuPortsURL = "http://ports.ubuntu.com/ubuntu-ports/"
+const ubuntuProURL = "https://esm.ubuntu.com/"
+
+// keep it sorted
+var validPro = []string{
+	"fips",
+	"fips-updates",
+}
+
+var ErrNoArchiveCredentials = errors.New("no archive credentials")
+
+func initProArchive(pro string, archive *ubuntuArchive) error {
+	if i := sort.SearchStrings(validPro, pro); !(i < len(validPro) && validPro[i] == pro) {
```

Comment:
This is subjective, but to simplify the condition, would it be cleaner to use `i >= len(validPro) || validPro[i] != pro`?